src/backend/optimizer/plan/aqumv.c

```diff
@@ -112,11 +111,11 @@ answer_query_using_materialized_views(PlannerInfo *root, RelOptInfo *current_rel
 
 	bool can_not_use_mv = (parse->commandType != CMD_SELECT) ||
 						  (parse->rowMarks != NIL) ||
-						  parse->hasAggs ||
 						  parse->hasWindowFuncs ||
 						  parse->hasDistinctOn ||
 						  /* Group By without agg could be possible though IMMV doesn't support it yet. */
 						  (parse->groupClause != NIL) ||
```

Comment:
also groupingSets ?

Reply:
how about add a test case ?

Reply:
Add case for we can't use mv when there is GroupingSets?
That's pointless, and we are working on Group By/Grouping Sets support, even cases added, they would be removed soon.

A quick practice will solve your doubts:

<img width="1790" alt="image" src="https://github.com/cloudberrydb/cloudberrydb/assets/17311022/5ea9e55a-6539-4f39-886f-e8cdefde80ff">